### PR TITLE
Hotfix/block type inline style override

### LINKED ISSUE #228: Current inline styles are lost when block type is changed
I noticed this when integrating Draft.js into my own application. In my case, I was setting per-block alignment based on user input by changing the block type using `RichUtils.toggleBlockType`. When I do this, if there are any existing currently inline styles, they are lost.

I can reproduce the same behavior on the [live Draft.js demo](https://facebook.github.io/draft-js/). Steps to reproduce:

1. Click bold and or italic.
2. Click the UL or H1 or BlockQuote button (or any other button that changes the block type). 

Notice the bold/italic style has been reverted. This does _not_ occur if you type some text in between steps 1 and 2. I attempted to trace the exact issue in the code, but due to my relative unfamiliarity with the codebase, I was unable to locate the specific bug or determine a suitable workaround. If you have any advice regarding a workaround, that would be most helpful.

Thanks much for a spectacular library!

### PR DESCRIPTION
**Pull request by rnicholus**
This fixes #228. I also fixed a few existing lint violations and added a simple but useful test to verify this behavior change.

### CODE REVIEW

File: src/model/immutable/EditorState.js [hunk 1]
```diff
@@ -377,16 +377,22 @@ class EditorState {
...
       currentContent: newContent,
       directionMap,
       undoStack,
       redoStack: Stack(),
       lastChangeType: changeType,
       selection: contentState.getSelectionAfter(),
-      forceSelection,
-      inlineStyleOverride: null,
-    });
+      forceSelection
```
Comment (hellendag) [inline src/model/immutable/EditorState.js]: Trailing comma

File: src/model/immutable/EditorState.js [hunk 2]
```diff
@@ -377,16 +377,22 @@ class EditorState {
...
+    };
+
+    // don't override inline style on block type change
+    if (changeType !== 'change-block-type') {
```
Comment (hellendag) [inline src/model/immutable/EditorState.js]: Could probably include `'adjust-depth'` here too.

**Comment by hellendag** [comment]
Thanks!

File: src/model/immutable/EditorState.js (hunk 1 above)
Comment (rnicholus) [inline src/model/immutable/EditorState.js]: This was pre-existing code, but I can certainly remove this. Note that trailing commas are legal in ES3+.

Comment (hellendag) [inline src/model/immutable/EditorState.js]: Oh, no, I'm saying that `forceSelection` can keep its comma (as `inlineStyleOverride` did previously).

Comment (rnicholus) [inline src/model/immutable/EditorState.js]: Ah, sorry. I'll re-add that then.

File: src/model/immutable/EditorState.js (hunk 2 above)
Comment (rnicholus) [inline src/model/immutable/EditorState.js]: I can certainly account for this case as well. It sounds like "adjust-depth" is used to represent tabs. Is this correct?

**Comment by rnicholus** [comment]
@hellendag My latest commit seems to address your review comments.

**Comment by hellendag** [comment]
> I can certainly account for this case as well. It sounds like "adjust-depth" is used to represent tabs. Is this correct?

Correct. Thanks for adding it.

**Comment by hellendag** [comment]
@facebook-github-bot import

**Comment by hellendag** [comment]
I'm going to merge this with a handful of changes to appease Flow. :)

**Pull request closed without merging**